### PR TITLE
fs-1821-Creating-an-application-in-Welsh

### PR DESCRIPTION
**Pull request by gio-karunakaran**

### CODE REVIEW

File: db/migrations/versions/a0b9e8f859a9_.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+revision = "a0b9e8f859a9"
+down_revision = "5bf853808dfb"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("applications", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("language", sa.String(), nullable=False))
```
Comment (gidsg) [inline db/migrations/versions/a0b9e8f859a9_.py:22]: This isn't going to work when deployed, as existing columns will be `null` i.e. they won't be `en` (as we've set currently in the test data.)

Instead, I think this needs to be `nullable=True`, and in the application code, assume `null` to be `en`

File: tests/test_routes.py [hunk 2]
```diff
@@ -85,6 +89,7 @@ def test_create_application_creates_unique_reference(
         "account_id": "usera",
         "fund_id": "47aef2f5-3fcb-4d45-acb5-f0152b5f03c4",
         "round_id": "c603d114-5364-4474-a0c4-c41cbf4d3bbd",
+        "language": "en",
```
Comment (gidsg) [inline tests/test_routes.py:92]: See previously comment, this won't happen on production, so the test data should include some `null` values and the application should still work and default in that case to `en`.

**Review by gidsg — CHANGES_REQUESTED** [review]

File: tests/test_routes.py (hunk 2 above)
Comment (gio-karunakaran) [reply tests/test_routes.py:92]: have update test data to reflect all cases of en, null and cy - will add logic to pull english forms if language = null and add more tests to cover this scenario 👍🏾

File: tests/test_routes.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ def test_create_application_is_successful(client):
         "account_id": "userb",
         "fund_id": "fund-b",
         "round_id": "summer",
-        "language": "en",
+        "language": "null",
```
Comment (gidsg) [inline tests/test_routes.py:39]: Think this might need to be `None` test the null database case?

Comment (gio-karunakaran) [reply tests/test_routes.py]: changed to None 👍🏾

File: db/migrations/versions/1147a99f5cc8_.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+revision = "1147a99f5cc8"
+down_revision = "a0b9e8f859a9"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("applications", schema=None) as batch_op:
+        batch_op.alter_column(
```
Comment (gidsg) [inline db/migrations/versions/1147a99f5cc8_.py:22]: Am I right in understanding this is just changing the type from `String` to `VARCHAR` if so can we delete this migration and just update the first one?

Comment (Adam-W1) [reply db/migrations/versions/1147a99f5cc8_.py:22]: Makes sense to me, is there any another reason this is split out @gio-karunakaran? 👀

Comment (Adam-W1) [reply db/migrations/versions/1147a99f5cc8_.py]: I made language an Enum and got rid of the two migrations in place of one

File: db/models/aggregate_functions.py [hunk 5]
```diff
@@ -215,12 +215,11 @@ def update_application_and_related_form(
     application.last_edited = func.now()
     form_sql_row = FormsMethods.get_form(application_id, form_name)
     # updating project name
-    if form_name == "project-information":
+    if form_name == "project-information" or "gwybodaeth-am-y-prosiect":
```
Comment (gidsg) [inline db/models/aggregate_functions.py:218]: Since we're getting the form in any case could we just lookup the key and not check the `form_name` as the key should be unique?

File: db/migrations/versions/1147a99f5cc8_.py (hunk 4 above)
Comment (gio-karunakaran) [reply db/migrations/versions/1147a99f5cc8_.py]: that makes sense to me! 👍🏾

File: db/models/aggregate_functions.py (hunk 5 above)
Comment (gio-karunakaran) [reply db/models/aggregate_functions.py:218]: We do not get a key for the form name, we only get keys for the questions/ fields, so we have to validate the form name manually (in english and welsh)

Comment (gidsg) [reply db/models/aggregate_functions.py:218]: Can we add a TODO to extract this update title logic into a separate function?

Comment (gio-karunakaran) [reply db/models/aggregate_functions.py:218]: the project name update logic has now been moved to a separate function (seemed much easier just to do it here than create a separate BAU and change down the line as it was a quick copy paste pretty much 👍🏾)

Comment (Adam-W1) [reply db/models/aggregate_functions.py:218]: So the reasoning here is that the form_name comes in as metadata from the [formrunner](https://github.com/communitiesuk/funding-service-design-application-store/blob/main/api/routes/application/routes.py#:~:text=%22form_name%22%3A%20request_json%5B%22metadata%22%5D.get(%22form_name%22)%2C). We only want to search for the field if the form that contains that field is being updated.

File: db/models/aggregate_functions.py [hunk 6]
```diff
@@ -214,24 +214,26 @@ def update_application_and_related_form(
...
+    update_project_name(form_name, question_json, application)
+    form_sql_row.json = question_json
+    update_statuses(application_id, form_name, is_summary_page_submit)
+    db.session.commit()
+    current_app.logger.info(
+        f"Application updated for application_id: '{application_id}."
+    )
+
+
+def update_project_name(form_name, question_json, application):
```
Comment (gidsg) [inline db/models/aggregate_functions.py:226]: Can we move this up so it's defined before it is referenced?

Comment (gio-karunakaran) [reply db/models/aggregate_functions.py]: have moved this up 👍🏾

**Review by gidsg — APPROVED** [review]
LGTM 👍
